src/tags/tests/service.rs

```diff
@@ -854,3 +854,135 @@ mod pass_tags_to_children_tests {
         cleanup();
     }
 }
+
+mod imply_all_ancestor_tags_tests {
+    use crate::repository::open_connection;
+    use crate::tags::repository as tag_repository;
+    use crate::tags::service::{get_tags_on_file, imply_all_ancestor_tags};
+    use crate::test::*;
+
+    #[test]
+    fn properly_implies_tags_in_depth_first_order() {
```

Comment:
This test doesn't actually test that depth first order is achieved. Needs to test if A+1/B+1/C, +1 is applied from B, not A. 

However, you already have that case handled in another test. Rename this test case to reflect what the test is actually testing